torch/csrc/autograd/engine.cpp

```diff
@@ -152,15 +152,21 @@ struct ReadyQueue {
 // the historic behavior.
 
 int NodeTask::getReentrantDepth() const {
-  return base_->reentrant_depth_;
+  std::shared_ptr<GraphTask> graph_task;
+  TORCH_INTERNAL_ASSERT(
+      graph_task = base_.lock(), "GraphTask is no longer valid!")
```

Comment:
No, please do not do this. Although it is not the case now, we reserve the right to completely eliminate `TORCH_INTERNAL_ASSERT`. If we did that, you wouldn't get the graph task at all in that case.

Reply:
Thinking about the error condition here, probably once we start processing a NodeTask, we should maintain liveness on the GraphTask. This implies to me that we shouldn't have this method on NodeTask at all; we should just ask for reentrant depth directly from our (live) GraphTask. Whether or not we should fix this in the PR, I need to read more to see.

Reply:
One of the other concerns here is performance since `getReentrantDepth` is called as part of `CompareNodeTaskTime` which is the comparison method of the heap. This method would be called multiple times for each `push`. I'm not sure of a clean way to resolve this issue yet.